src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -224,6 +225,7 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 		private _container: HTMLElement | undefined,
 		private _shellLaunchConfig: IShellLaunchConfig,
 		@ITerminalInstanceService private readonly _terminalInstanceService: ITerminalInstanceService,
+		@ITerminalService private readonly _terminalService: ITerminalService,
```

Comment:
This is no good because then TerminalService depends on TerminalInstance which depends on TerminalService. This was actually the main reason TerminalInstanceService was introduced. We should either move `preparePathForTerminalAsync` there or into a util file.

Reply:
`PreparePathForTerminalAsync` relies on` local/remoteTerminalService` and I thought we were trying to get rid of `TerminalInstanceService`. I'll go with that for now.

Reply:
hmm it was working before. with it now in terminalInstanceService, I'm seeing contributions.ts:123 Unable to instantiate workbench contribution StatusBarColorProvider. Error: taskService -> terminalService -> remoteTerminalService -> terminalInstanceService -> remoteTerminalService
    at InstantiationService._createAndCacheServiceInstance (/C:/Users/merogge/repos/vscode/out/vs/platform/instantiation/common/instantiationService.js:143)
    at 

bc I have to inject remoteTerminalService / local terminalService to get the wslPath

Reply:
oh looks like terminalInstanceService isn't actually used rn in remoteTerminalService